src/Microsoft.Restier.Core/Api.Interfaces.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Restier.Core
+{
+    [CLSCompliant(false)]
+    public interface IApiConfigurator
+    {
+        void Configure(IServiceCollection services, Type apiType);
+    }
+
+    public interface IApiInitializer
+    {
+        void Initialize(ApiConfiguration configuration);
+    }
+
+    public interface IApiContextConfigurator
+    {
```

Comment:
Public methods in ApiConfigurationAttribute are grouped into these 3 interfaces, to run some customized logic in various stage.